index.js

```diff
@@ -144,12 +148,16 @@ function mirror (src, dst, opts, cb) {
 
     var rs = a.fs.createReadStream(a.name)
     var ws = b.fs.createWriteStream(b.name, {mode: a.stat.mode})
+    var increment = through(function (chunk, enc, cb) {
+      progress.emit('chunk', chunk)
+      cb(null, chunk)
+    })
 
     rs.on('error', onerror)
     ws.on('error', onerror)
     ws.on('finish', cb)
 
-    rs.pipe(ws)
+    rs.pipe(increment).pipe(ws)
```

Comment:
technically through2 is not needed here as we don't need backpressure. you can actually just attach a data listener on `rs`. 

``` js
rs.pipe(ws)
rs.on('data', function (data) {
  progress.emit(...)
})
```